src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -134,27 +154,38 @@ public static class TagHelperBlockRewriter
                     SourceLocation symbolStartLocation;
 
                     // Check for attribute start values, aka single or double quote
-                    if (IsQuote(htmlSymbols[i + 1]))
+                    if ((i + 1) < htmlSymbols.Length && IsQuote(htmlSymbols[i + 1]))
                     {
                         // Move past the attribute start so we can accept the true value.
                         i++;
-                        symbolStartLocation = htmlSymbols[i + 1].Start;
+                        symbolStartLocation = symbol.Start;
```

Comment:
sorry, I was wrong here.  `symbol != htmlSymbols[i]` anymore.  you were right to spell `htmlSymbols[i].Start` out.  (if I'd been correct, would have worked to assign `symbolStartLocation` above the `if`.)